Octal (#1740)
Codecademy/docs: content/general/concepts/octal/octal.md

---
Title: 'Octal' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
Description: 'Octal is a number system that uses digits from 0 to 7. Compared to the decimal system each digit corresponds to the power of 8 instead of the power of 10.' # Required; ideally under 150 characters and starts with a present-tense verb (used in search engine results and content previews)
Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
  - 'Computer Science'
  - 'Data Science'
  - 'Code Foundations'
Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!

>   - 'Best Practices'

Reviewer: "Best Practices" seems an odd tag for this. Wouldn't something like "Arithmetic" make more sense?

  - 'Encoding'
CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
  - 'paths/code-foundations'
  - 'paths/computer-science'
---

The **octal** number system can be compared with the decimal, the hexadecimal, the binary, and many other systems. As the name says the system consists of 8 different digits (0-7). In our daily life, we are used to the decimal system, which consists of steps to the power of 10. Instead the Octal system uses steps to the power of 8.

However in modern technologies and programming the octal system is rarely used, instead, the binary or hexadecimal system is used commonly.
In the case of the hexadecimal system, the octal system has the advantage of using only numbers, therefore, renouncing using the alphabet.

## Example

> octal	decimal

Reviewer: This should be a Markdown table.

---
Title: 'Octal' 

> Description: 'Octal is a number system that uses digits from 0 to 7. Compared to the decimal system each digit corresponds to the power of 8 instead of the power of 10.' 

Reviewer 2: This description should be rewritten.

> Description: 'Octal is a number system base 8. It consists of figures from 0 to 7. Every figure in a number written in octal corresponds to 8 to the power of the certain digit.'

Reviewer 2: ```suggestion
Description: 'Octal is a number system base 8.'
```

Subjects: 
  - 'Computer Science'
  - 'Data Science'
  - 'Code Foundations'
Tags: 
  - 'Arithmetic'
  - 'Encoding'
CatalogContent: 
  - 'paths/code-foundations'
  - 'paths/computer-science'
---

> The **octal** number system is base 8. The number system that we are used to is decimal, which is base 10. There are other number systems like [binary](https://www.codecademy.com/resources/docs/general/binary) or hexadecimal that work similarly. The octal system being base 8 means that there are only 8 possible digits: 0-7. Nevertheless, every number can be displayed.

Reviewer 2: ```suggestion
The **octal** number system is a base-8 number system that uses digits from 0-7 to represent any number. Octal is a number system similar to other number systems like [binary](https://www.codecademy.com/resources/docs/general/binary) and hexadecimal.
```

> The **octal** number system can be compared with the decimal, the hexadecimal, the [binary](https://www.codecademy.com/resources/docs/general/binary), and many other systems. As the name says the system consists of 8 different digits (0-7). In our daily life, we are used to the decimal system, which consists of steps to the power of 10. Instead the Octal system uses steps to the power of 8.

Reviewer 2: This sentence should be reworded to include the definition of the octal system.

Reviewer 2: Something like: "The **octal** number system is a number system that is base 8. The traditional number system that we are used to is decimal which is base 10."

> In the past, the octal system was used more often because systems were based on 12/24/36-bit but in today's technologies, especially with 16/32/64-bit the binary or hexadecimal system is used commonly. In the case of the hexadecimal system, the octal system has the advantage of using only numbers, therefore, renouncing using the alphabet.

Reviewer 2: ```suggestion
The octal system was often used in the past because systems were based on 12, 24, or 36 bits. The binary or hexadecimal system is commonly used in today's technologies, especially with 16, 32, or 64-bit machines. The octal number system has an advantage over the hexadecimal system in that we do not need to use characters from the alphabet to express a number in octal.
```

> The octal system was often used in the past because systems were based on 12, 24, or 36 bits. The binary or hexadecimal system is commonly used in today's technologies, especially with 16, 32, or 64-bit machines. The octal number system has an advantage over the hexadecimal system in that we do not need to use characters from the alphabet to express a number in octal.

Reviewer 2: ```suggestion
Octal was often used because machines were based on 12, 24, or 36 bits. In contrast, binary or hexadecimal are commonly used in modern technologies, especially with 16, 32, or 64-bit machines. The octal number system has an advantage over the hexadecimal system in that characters from the alphabet are not used to express a number in octal.
```

## Example

octal|decimal
---|---
0|0
1|1
2|2
3|3
7|7
10|8
11|9
12|10
20|16
40|32
100|64

> Each digit is a power of 8, so the last digit (decimal 0-7) is 8^0, equalling 1, the one in front of it 8^1, and so on. If we write the octal system as a calculation to convert it to the decimal system we would get:

Reviewer 2: This sentence should be rewritten to clearly express how an octal number is related to a decimal number.  Also, use the \<sup\> and \<\/sup\> Markdown tags to write exponents.

```

> 125 = 1 * 8^2   + 2 * 8^1   + 5 * 8^0

Reviewer 2: As this example is not code, it should not be in a code block.

Reviewer: @HishamT It was my idea for the code block so the spacing worked out. I suppose it would work without the spacing, it would still get the point across.

Author: I think with the changes I made now it should be understandable without the code block.

> Each digit corresponds to a power of 8, so the last digit (decimal 0-7) is 8<sup>0</sup>, equalling 1, the one in front of it 8<sup>1</sup>, and so on. So each digit represents the figure times 8 to the power of its position. If we write the octal system as a calculation to convert it to the decimal system we would get:

Reviewer 2: ```suggestion
Each digit corresponds to a power of 8. The last digit (decimal 0-7) is 8<sup>0</sup>, equalling 1. The one in front of it is 8<sup>1</sup>, and so on. So each digit represents the figure times 8 to the power of its position. An octal number is converted to decimal like so (using octal 125 as an example):
```

    = 64        + 16        + 5
    = 85

> 

Reviewer 2: Remove these whitespaces to not cause a formatting error.
